src/lib/dai.c

```diff
@@ -48,7 +48,7 @@ static struct dai_group *dai_group_find(uint32_t group_id)
 {
 	struct list_item *dai_groups;
 	struct list_item *group_item;
-	struct dai_group *group;
+	struct dai_group *group = NULL;
```

Comment:
we could do
```
	list_for_item(group_item, dai_groups) {
		struct dai_group *group = container_of(group_item, struct dai_group, list);

		if (group->group_id == group_id)
			return group;
	}
	return NULL;
```
and remove the top level "group" variable completely

Reply:
It's against our code guidelines, all variables need to be declared at the top of the function.